src/NFCManager.cpp

```
@@ -2682,32 +2680,62 @@ bool NFCManager::isWriteQueueEmpty() const {
+
+        // Blocks carrying identity and print-critical values must all be
+        // present: a partial read would publish a plausible-looking spool with
+        // the wrong colour, weight, or nozzle/bed temperatures — worse than no
+        // read at all. Stop at the first essential failure instead of working
+        // through the rest; on a shared SPI bus each further attempt can block
+        // for the bus-guard timeout, and the whole read is discarded anyway.
+        // The remaining blocks (4 unparsed, 13 date, 14 length, 16 extended
+        // colour) degrade to zeros without misrepresenting the filament.
+        if (failed) {
+            if (bambuIsEssentialIndex(i)) {
+                Serial.printf("NFCManager: Bambu essential block %d missing, discarding read\n", blockNo);
```

Clang (14.0.6)

[warning] do not call c-style vararg functions

(cppcoreguidelines-pro-type-vararg)